// lib/es6/mm/payment.js
// Generated by BUCKLESCRIPT VERSION 2.2.3, PLEASE EDIT WITH CARE

import * as Block from "bs-platform/lib/es6/block.js";
import * as Caml_builtin_exceptions from "bs-platform/lib/es6/caml_builtin_exceptions.js";

function billPayment(param_0) {
  return /* BillPayment */Block.__(0, [param_0]);
}

function otherPayment(param_0) {
  return /* OtherPayment */Block.__(1, [param_0]);
}

function make(cat, amt, date, filled, about) {
  return {
          id: (((Math.random() * 1e10) | 0)),
          category: cat,
          amount: amt,
          date: date,
          filled: filled,
          about: about
        };
}

function getBill(t) {
  var match = t.about;
  if (match.tag) {
    throw [
          Caml_builtin_exceptions.failure,
          "Cannot get bill: wrong type of payment"
        ];
  } else {
    return match[0];
  }
}

function getRecipient(t) {
  var match = t.about;
  if (match.tag) {
    return match[0];
  } else {
    return match[0][/* owedTo */3];
  }
}

export {
  billPayment ,
  otherPayment ,
  make ,
  getBill ,
  getRecipient ,
  
}
/* No side effect */
